bubble_sort.s

```diff
@@ -0,0 +1,234 @@
+# Copyright (C) 2018 Yauheni Sharamed
+# This program is a part of MIPT-MIPS source code
+# distributed under terms of MIT License
+#
+# MIPT-MIPS web site: http://mipt-ilab.github.io/mipt-mips/
+#
+# This is realisation of basic bubble sort algorithm.
+# Prototype: void bubble_sort(int* arr , int num)
+# num - number of elements in array, num < 536870911
+# arr - pointer to the array
+# Elements of array could be signed integers
+# Both arguments must be sent in $a0 (arr) and $a1 (num)
+
+
+
+.globl main
+.data
+ 	numberofelem: .word numberofelem_data
+	insertarr: .word insertarr_data
+	resbody: .word resbody_data
+	space: .word space_data
+	badnum: .word badnum_data
+	printarr: .word printarr_data
+ 
+	numberofelem_data: .asciiz "Number of elements in array: "
+	insertarr_data: .asciiz "Elements of array:"
+	resbody_data: .asciiz "Sorted array: "
+	space_data: .asciiz " "
+	badnum_data: .asciiz "Bad number of elements: it must be positive integer. "
+	printarr_data: .asciiz "Input array :"
+	
+	is_static: .word 1
+	num: .word 6
+	arr: .word 6 , 3 , -1 , 3 , 0 , 112
+.text
+main:
+# Because of lack of scanf function and sbrk syscall in current version, it is necessary to use
+# static array (arr) and predefined number of elements (num).
+# If is_static != 0 then the static block is executed
+# If is_static == 0 then the block with scanf and memory allocation is executed
+
```

Comment:
I suggest to add a `li $sp, 0x30000` here, because we do not initialize stack pointer yet, and stores/loads around zero look suspicious.

Reply:
To be honest, I don't completely understand, why do we initialize `$sp` with zero?

Reply:
Usually initialization is performed by OS, and we do not have any in current modeling flow mode.